File: swift/one.swift
print("Введите число S: ", terminator: "")
let S = Int(readLine()!)!
print("Введите число N: ", terminator: "")
let N = Int(readLine()!)!
print("Введите число M: ", terminator: "")
let M = Int(readLine()!)!
func days(S: Int, N: Int, M: Int) -> Int{
    if M > N {
        return -1
    }
    let sundays = S / 7
    let day_work = S - sundays
    let need_food = S * M
    let min_days = (need_food + N - 1) / N
    if min_days > day_work {
        return -1
    }
    return min_days
}
let result = days(S: S, N: N, M: M)
if result == -1{
print("Выжить невозможно")
} else{
    print("Минимальное количество дней: \(result)")
}

File: Java/one.java
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Введите число S: ");
        int S = scanner.nextInt();
        System.out.print("Введите число N: ");
        int N = scanner.nextInt();
        System.out.print("Введите число M: ");
        int M = scanner.nextInt();
        int result = days(S, N, M);
        if (result == -1) {
            System.out.println("Выжить невозможно");
        } else {
            System.out.println("Минимальное количество дней: " + result);
        }
        scanner.close();
    }
    public static int days(int S, int N, int M) {
        if (M > N) {
            return -1;
        }
        int sundays = S / 7;
        int foodNeed = S * M;
        int shopWorkDays = S - sundays;
        int minDays = (foodNeed + N - 1) / N;
        if (minDays > shopWorkDays) {
            return -1;
        }
        return minDays;
    }
}
